### ISSUE TITLE
Allow choosing strictness: no redundant left pizza, or no left pizza at all

### ISSUE THREAD

**Issue by zwilias** [truqu/elm-review-noleftpizza#2]
Currently the rule does not allow _any_ use of `<|`. The rule would definitely be useful to more people if it only disallowed redundant left pizza's. In other words, left pizza where removing the pizza does not require adding any parenthesis.

However, I'm wondering how to "structure" the rule.
Currently, we have `NoLeftPizza.rule`. 

The two alternatives that come to mind:

- add a `NoLeftPizza.Redundant.rule`
- add a configuration parameter: `NoLeftPizza.rule : Any | Redundant -> Rule`

I'm leaning towards the latter for clarity, but it does mean a breaking change. Which isn't a big deal at all, but it might be worth considering if any more variations are likely to show up in the future.

CC @Janiczek who might be interested in this

**Comment by Janiczek** [comment]
cc @jhrcek

FWIW @zwilias both options are fine with me

**Comment by jfmengels** [comment]
I think a configuration parameter would be better here.

I think it would also be useful to have an exception to this rule for tests (configurable, or maybe baked in?).

Also, I heard that `elm-ui` and `elm-animator` code make heavy use of left-pizza. But that is probably too much configuration for this rule.

Anyway, I would personally like to use the rule with the redundant option :slightly_smiling_face:

**Comment by zwilias** [comment]
I went with the configuration parameter.

I did not build in any tests/elm-ui/elm-animator/... code at this time. Exempting tests can already happen through whitelisting the `tests` folder. That doesn't quite cut it for inline tests, but those seem less common. Either way, I'd consider that to be a separate issue, as built in support for other libraries would be. Thanks for the suggestions and input! <3
